include/vcpkg/base/expected.h

```diff
@@ -198,25 +200,25 @@ namespace vcpkg
             }
         }
 
-        template<class F, class U = map_t<F>>
-        U then(F f) const&
+        template<class F, class... Args, class U = std::invoke_result_t<F, const_ref_type, Args&&...>>
+        U then(F f, Args&&... args) const&
         {
             if (has_value())
             {
-                return f(*m_t.get());
+                return std::invoke(f, *m_t.get(), static_cast<Args&&>(args)...);
```

Comment:
Doesn't this require C++17? Can we actually use C++17?

Reply:
I assume that with the new "ship binaries" approach, as long as it passes CI it should be sufficient.